docker/requirements.txt

```diff
@@ -19,7 +19,10 @@ libsass
 pip
 pylint
 pymysql
-pytest
+pytest-html>=1.20,<1.21
+pytest-xdist>=1.28,<1.29
+pytest>=4.4,<4.5
```

Comment:
Two things:

1. Hmm.  My intention wasn't to put everything in one environment, but to put common things in this environment, that is, things used by many/most downstream users of base.

   We will deploy off base.  Therefore, I don't think the testing stuff should go here.  In the ci2 image, I have an intermediate image with the stuff necessary to test Hail and other services.

   By the same argument, twine is only used for deploy so should go in the deploy image.

   I'm opening to throwing everything in the same image (easy), or waiting until ci2 (easier to manage than the current setup), but I'd like to discuss it first.  Thoughts?

2. This isn't pinning.  I think this file should be reproducible and we should put the exact version here.

Reply:
I'm fine with waiting for ci2.

re: pinning, I see the value for our deployed artifact.

I maintain that for the PyPI package we should accept any sem ver compatible version so as to ease dependency hell for our users.